tests/learn-promotion.sh

```
@@ -0,0 +1,107 @@
+# 3. THE POINT: a harmful session PINS the entry at LOW even at seen>=3.
+S="$WORK/s-harm"; build_store "$S" outcome a1:helpful b2:helpful c3:harmful; merge_sessions "$S" a1 b2 c3
+ok "outcome-mode: one harmful session pins at LOW despite seen>=3" "[ \"$(conf_of "$S")\" = LOW ]"
+
+# 4. Unresolved is NOT harmful and NOT promotable — absence of evidence stays LOW (resolver RULE 29).
+S="$WORK/s-pend"; build_store "$S" outcome zz:helpful; merge_sessions "$S" a1 b2 c3
+ok "outcome-mode: unresolved sessions stay LOW (never HIGH-by-repetition)" "[ \"$(conf_of "$S")\" = LOW ]"
+
+# 5. FAIL-OPEN, LOUDLY: outcome mode with no sidecar degrades to seen-mode AND says so on stderr.
+S="$WORK/s-nosidecar"; build_store "$S" outcome; merge_sessions "$S" a1 b2 c3
+ok "no sidecar: falls back to seen-mode (reaches HIGH, not silently pinned LOW)" "[ \"$(conf_of "$S")\" = HIGH ]"
+ok "no sidecar: the fallback is ANNOUNCED on stderr, never silent" "grep -q 'LEARN-PROMOTION' "$WORK/err-c3.txt""
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)